providers/src/airflow/providers/google/cloud/operators/gcs.py

```diff
@@ -794,7 +799,10 @@ def execute(self, context: Context) -> list[str]:
             orig_start = context["data_interval_start"]
             orig_end = context["data_interval_end"]
         except KeyError:
-            orig_start = pendulum.instance(context["execution_date"])
+            if AIRFLOW_V_3_0_PLUS:
+                orig_start = pendulum.instance(context["logical_date"])
+            else:
+                orig_start = pendulum.instance(context["execution_date"])  # type: ignore[typeddict-item]
```

Comment:
`logical_date` is in the context on Airflow 2 too, so we can always just use that

```suggestion
            orig_start = pendulum.instance(context["logical_date"])
```

Reply:
Changed it